src/main/java/ca/bc/gov/chefs/etl/forms/pcd/upcc/pcPatientServices/processor/PcdUpccPCPSApiResponseProcessor.java

```diff
@@ -0,0 +1,144 @@
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public void process(Exchange exchange) throws Exception {
+    String payload = exchange.getIn().getBody(String.class);
+    payload = JsonUtil.roundDigitsNumber(payload);
+    ObjectMapper mapper = new ObjectMapper();
+    List<Root> pcpsModels = mapper.readValue(payload, new TypeReference<List<Root>>() {
+    });
+    List<PCPSUpccSubmission> parsedUpccPCPS = parsePCPSRequest(pcpsModels);
+
+    // validateRecordCount(pcpsModels, parsedUpccPCPS);
+
+    List<IModel> iModels = (List<IModel>) (List<?>) parsedUpccPCPS;
+    Map<String, List<List<String>>> map = CSVUtil.provider(iModels);
+
+    boolean isHeaderAdded = (boolean) exchange.getProperties().get(Constants.IS_HEADER_ADDED);
+    List<String> filesGenerated = FileUtil.writeToCSVFile(map, PCDConstants.PCD_UPCC_PCPS_DIR, isHeaderAdded);
+
+    SuccessResponse successResponse = new SuccessResponse();
+    successResponse.setFiles(filesGenerated);
+    exchange.getIn().setBody(mapper.writeValueAsString(successResponse));
+  }
+
+  private List<PCPSUpccSubmission> parsePCPSRequest(List<Root> pcpsModels)
+      throws IllegalArgumentException, IllegalAccessException, NoSuchFieldException, SecurityException {
+    List<PCPSUpccSubmission> parsedUpccPCPS = new ArrayList<>();
+    for (Root root : pcpsModels) {
+      PCPSUpccSubmission pcpsSubmission = new PCPSUpccSubmission();
+
+      pcpsSubmission.setSubmissionId(root.getForm().getSubmissionId());
+      pcpsSubmission.setCreatedAt(CSVUtil.formatDate(root.getForm().getCreatedAt()));
+      pcpsSubmission.setLateEntry(root.getLateEntry());
+      pcpsSubmission.setSubmitterFullName(root.getForm().getFullName());
+      pcpsSubmission.setSubmitterUserName(root.getForm().getUsername());
+      pcpsSubmission.setSubmitterEmail(root.getForm().getEmail());
+      pcpsSubmission.setSubmissionStatus(root.getForm().getStatus());
+      pcpsSubmission.setSubmissionVersion("" + root.getForm().getVersion());
+      pcpsSubmission.setSubmissionFormName(root.getForm().getFormName());
+      pcpsSubmission.setUpccName(root.getUpccName());
+      pcpsSubmission.setPcnCommunityName(root.getPcnCommunity());
+      pcpsSubmission.setHealthAuthority(root.getHealthAuthority());
+      pcpsSubmission.setUpccTypeOfCare(root.getUpccTypeOfCare());
+      pcpsSubmission.setFiscalYear(root.getFiscalYear());
+      pcpsSubmission.setPeriodReported(root.getPeriodReported());
+      pcpsSubmission.setReasonForExceptPeriodRep(root.getReasonForExceptionInPeriodReported());
+      pcpsSubmission.setNotes(root.getDataSubmission().getNotes());
+
+      List<PCPSUpccSubmissionData> pcpsSubmissionDataList = new ArrayList<>();
+
+      for (Integer i = 0; i < 13; i++) {
+        PCPSUpccSubmissionData pcpsSubmissionData = new PCPSUpccSubmissionData();
+        Object o = root.getDataSubmission(); // The object you want to inspect
+        Class<?> c = o.getClass();
+        // Check if all fields are null then skip the record
```

Comment:
See previous comments about index variables, field access, etc.

Reply:
Removed Class c as it is not needed, updated o's name to be more descriptive.